src/commands/statusline.ts

```
@@ -64,6 +76,26 @@ export const statuslineCommand = define({
 		// Extract session ID from hook data
 		const sessionId = hookData.session_id;
 
+		// Create semaphore instance for statusline
+		const semaphore = SemaphoreFactory.statusline(ctx.values.refreshInterval);
+
+		// Rate limiting check
+		if (!ctx.values.disableCache) {
```

ESLint

[error] Unexpected any value in conditional. An explicit comparison or type conversion is required.

(ts/strict-boolean-expressions)